### PR TITLE
Watershed Report - First Pass

### PR DESCRIPTION
**Pull request by LiamF-FS**
# Description

Initialize Watershed Report.

This is a first pass and required further work.

## Types of changes

New feature (non-breaking change which adds functionality)

## Further comments

We need to investigate behaviour for various regions, lakes, etc. 

We need to handle edge cases of missing data, and ensure the report displays the expected values for the region of interest.

To test this out, load the watersheds module, and click a point on the map. Some points will not have data - however, the points that do have data, you should see a watershed after clicking on the map, and hitting view report should generate a full report for that watershed.

### CODE REVIEW

File: client/src/utils/api.js [hunk 1]
```diff
@@ -24,18 +24,15 @@ export const getAllWatershedLicences = async () => {
     return await requestWithErrorCatch(`${env.VITE_BASE_API_URL}/watershed/licences`);
 }
 
+export const getWatershedBySearch = async (wls_id) => {
```
Comment (LiamF-FS) [inline client/src/utils/api.js:27]: This should be titled getWatershedLicenceBySearch

Comment (LiamF-FS) [reply client/src/utils/api.js:27]: another route should be mate for watershed search. I will update

File: backend/queries/watershed/get_licence_import_dates.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+get_licence_import_dates_query = f"""
+    SELECT
+        dataset,
+        import_date
+    FROM
+        bcwat_lic.bc_data_import_date
```
Comment (Kashike-Umemura) [inline backend/queries/watershed/get_licence_import_dates.py:6]: You likely could filter this down since I put the hydat import date in this table as well. But it's also one row so no big deal imo

File: backend/queries/watershed/get_watershed_allocations_by_id.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+get_watershed_allocations_by_id_query = """
+	SELECT
+    	*
+    FROM
+		bcwat_lic.get_allocs_per_wfi(%(watershed_feature_id)s)
```
Comment (Kashike-Umemura) [inline backend/queries/watershed/get_watershed_allocations_by_id.py:5]: I know that this probably returns a lot of columns, but I think it would be better if we select the specific columns and not `*` because then we would know if removing a column breaks some functionality in this query.
If you keep it as `*`, we wouldn't know if a column goes missing as easily.

File: backend/queries/watershed/get_watershed_annual_hydrology_by_id.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+get_watershed_annual_hydrology_by_id_query = """
+    SELECT
+        *
+    FROM
+        bcwat_lic.get_annual_hydrology(%(watershed_feature_id)s);
```
Comment (Kashike-Umemura) [inline backend/queries/watershed/get_watershed_annual_hydrology_by_id.py:5]: Same as before

File: backend/queries/watershed/get_watershed_industry_allocations_by_id.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+get_watershed_industry_allocations_by_id_query = """
+    SELECT
+        *
+    FROM
+        bcwat_lic.get_allocs_by_industry(%(watershed_feature_id)s);
```
Comment (Kashike-Umemura) [inline backend/queries/watershed/get_watershed_industry_allocations_by_id.py:5]: Same here, but might be less relevant since everything gets aggregated in to a JSON obj

File: backend/queries/watershed/get_watershed_monthly_hydrology_by_id.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+get_watershed_monthly_hydrology_by_id_query = """
+    SELECT
+        *
+    FROM
+        bcwat_lic.get_monthly_hydrology(%(watershed_feature_id)s, %(in_basin)s, %(region_id)s);
```
Comment (Kashike-Umemura) [inline backend/queries/watershed/get_watershed_monthly_hydrology_by_id.py:5]: Ditto

**Review by Kashike-Umemura — APPROVED** [review]
Overall looks great! I left some minor comments about the queries that don't need to be changed. But if a data model change could happen in the future, I think it would help to eliminate ambiguity

File: backend/routers/watershed.py [hunk 7]
```diff
@@ -96,7 +130,113 @@ def get_watershed_report_by_id(id):
             id (int): Watershed ID.
     """
 
-    watershed_metadata = app.db.get_watershed_report_by_id(watershed_feature_id=id)
-    # print(json.dumps(watershed_metadata, indent=2))
+    region = app.db.get_watershed_region_by_id(watershed_feature_id=id)
+    print(region)
```
Comment (AndersFarrFoundry) [inline backend/routers/watershed.py:134]: rm print

File: backend/utils/surface_water.py [hunk 8]
```diff
@@ -1,4 +1,6 @@
 import polars as pl
+import json
+from datetime import datetime
```
Comment (AndersFarrFoundry) [inline backend/utils/surface_water.py:3]: Are these imports now necessary? It looks like the change in this file is minimal. I've probably been removing a lot because my vs code removes unused imports on save.

**Review by AndersFarrFoundry — APPROVED** [review]
lgtm I don't know nearly as much as Kash about the SQL side of things! Successfully generated some reports though!

File: backend/utils/surface_water.py (hunk 8 above)
Comment (LiamF-FS) [reply backend/utils/surface_water.py:3]: probably not... I will take a look

File: backend/routers/watershed.py (hunk 7 above)
Comment (LiamF-FS) [reply backend/routers/watershed.py:134]: cheers!
